src/main/java/echoserver/ClientConnection.java

```diff
@@ -10,30 +10,37 @@ public ClientConnection(Socket socket) {
         this.socket = socket;
     }
 
-    public String promptInput() throws IOException {
-        lastSavedInput = convertInputStreamToString(socket.getInputStream());
-        return lastSavedInput;
+    public String readInput() throws IOException {
+        String connectionInput = convertInputStreamToString(socket.getInputStream());
+        return connectionInput;
     }
 
-    public String readLastSavedInput(){
-        return lastSavedInput;
+    public void writeInput(String input) throws IOException {
+        String messageFormat = "=> ";
+        String message = messageFormat + input + "\n";
+        byte[] messageData = message.getBytes();
+
+        socket.getOutputStream().write(messageData);
     }
 
-    public OutputStream getOutputStream() throws IOException{
-        return socket.getOutputStream();
+    public void writeConnectionClosingMessage() throws IOException{
+        String message = "Connection Closing...\n";
+        byte[] messageData = message.getBytes();
+
+        socket.getOutputStream().write(messageData);
```

Comment:
This is totally a personal preference, but I try to avoid some temporary variables in Java purely for aesthetic reasons. I always opt for something like this vs having to declare a variable with a type of array. 🤷‍♂ 

```suggestion
        socket.getOutputStream().write(messageData.getBytes());
```